coderd/notifications/provider.go

```diff
@@ -0,0 +1,44 @@
+package notifications
+
+import (
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/database"
+)
+
+type HandlerRegistry struct {
+	handlers map[database.NotificationMethod]Handler
+}
```

Comment:
Does this need to be safe for concurrent use?

Reply:
Technically yes but practically no.
The handlers are only ever mutated during the setup of the manager, but technically can be modified with the `WithHandlers` call afterwards, which they never are.